### PR TITLE
feat: separate static and upload files on cloud storage

### PR DESCRIPTION
**Pull request by lordmallam**

### CODE REVIEW

File: aether/sdk/management/commands/cdn_publish.py [hunk 1]
```diff
@@ -91,11 +91,18 @@ def handle(self, *args, **options):
         self.__publish(webpack_dir, storage_path)
 
     def __publish(self, local_dir, storage_path):
+        storage_backend = os.environ.get('DJANGO_STORAGE_BACKEND')
+        if storage_backend == 's3':
+            static_storage_class = get_storage_class('aether.sdk.conf.storages.StaticS3')()
+        elif storage_backend == 'gcs':
+            static_storage_class = get_storage_class('aether.sdk.conf.storages.StaticGCS')()
+        else:
+            static_storage_class = default_storage
```
Comment (obdulia-losantos) [inline aether/sdk/management/commands/cdn_publish.py]: `get_storage_class(settings.STATICFILES_STORAGE)` ???

File: aether/sdk/conf/settings.py [hunk 2]
```diff
@@ -658,7 +658,7 @@ def get_required(name):
             AWS_DEFAULT_ACL = get_required('AWS_DEFAULT_ACL')
 
             if bool(os.environ.get('COLLECT_STATIC_FILES_ON_STORAGE')):
-                STATICFILES_STORAGE = 'storages.backends.s3boto3.S3Boto3Storage'
+                STATICFILES_STORAGE = 'aether.sdk.conf.storages.StaticS3'
```
Comment (obdulia-losantos) [inline aether/sdk/conf/settings.py]: `STATIC_BUCKET_NAME = get_required('STATIC_BUCKET_NAME')`

File: aether/sdk/conf/settings.py [hunk 3]
```diff
@@ -667,7 +667,7 @@ def get_required(name):
             GS_BUCKET_NAME = get_required('BUCKET_NAME')
 
             if bool(os.environ.get('COLLECT_STATIC_FILES_ON_STORAGE')):
-                STATICFILES_STORAGE = 'storages.backends.gcloud.GoogleCloudStorage'
+                STATICFILES_STORAGE = 'aether.sdk.conf.storages.StaticGCS'
```
Comment (obdulia-losantos) [inline aether/sdk/conf/settings.py]: `STATIC_BUCKET_NAME = get_required('STATIC_BUCKET_NAME')`

File: aether/sdk/conf/storages.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+def get_required(name):
+    try:
+        return os.environ[name]
+    except KeyError as key:
+        raise RuntimeError(f'Missing {key} environment variable!')
+
+
+class StaticS3(S3Boto3Storage):
+    bucket_name = get_required('STATIC_BUCKET_NAME')
```
Comment (obdulia-losantos) [inline aether/sdk/conf/storages.py]: Why not `bucket_name = settings.STATIC_BUCKET_NAME`?

Comment (lordmallam) [reply aether/sdk/conf/storages.py]: because I am not setting `STATIC_BUCKET_NAME` on settings

Comment (obdulia-losantos) [reply aether/sdk/conf/storages.py]: See my comment above :stuck_out_tongue:

Comment (obdulia-losantos) [reply aether/sdk/conf/storages.py]: I mean, in the settings file

File: aether/sdk/management/commands/cdn_publish.py [hunk 5]
```diff
@@ -91,11 +91,13 @@ def handle(self, *args, **options):
         self.__publish(webpack_dir, storage_path)
 
     def __publish(self, local_dir, storage_path):
+        storage_backend = os.environ.get('DJANGO_STORAGE_BACKEND')
```
Comment (obdulia-losantos) [inline aether/sdk/management/commands/cdn_publish.py]: I think this line is not longer needed :thinking:

File: aether/sdk/management/commands/cdn_publish.py [hunk 6]
```diff
@@ -91,11 +91,12 @@ def handle(self, *args, **options):
         self.__publish(webpack_dir, storage_path)
 
     def __publish(self, local_dir, storage_path):
+        static_storage_class = get_storage_class(settings.STATICFILES_STORAGE)()
```
Comment (obdulia-losantos) [inline aether/sdk/management/commands/cdn_publish.py]: if you move this line outside the class, I think the tests will pass :crossed_fingers:

File: aether/sdk/tests/test_commands.py [hunk 7]
```diff
@@ -281,6 +281,13 @@ def test_cdn_publish(self, mock_cdn_save):
                      stdout=self.out,
                      stderr=self.out)
 
+        call_command('cdn_publish',
+                     '--cdn-url=http://cdn-server/path/to/',
+                     '-w=aether/sdk/tests/webpackfiles',
+                     '--storage-path=/__assets__/sdk/',
+                     stdout=self.out,
+                     stderr=self.out)
```
Comment (obdulia-losantos) [inline aether/sdk/tests/test_commands.py]: duplicated?

Comment (lordmallam) [reply aether/sdk/tests/test_commands.py]: to cover the if conditions for when cdn_url and storage_path end and start with `/` respectively

Comment (obdulia-losantos) [reply aether/sdk/tests/test_commands.py]: in that case I would split it in two tests with different values, you'll never know if the following check belongs to the first or the second call :woman_shrugging:

**Pull request merged**